lib/autoproj/cli/commit.rb

```diff
@@ -5,34 +5,48 @@
 module Autoproj
     module CLI
         class Commit < InspectionTool
-            def run(*user_selection, options = Hash.new)
-                pkg = manifest.main_package_set.create_autobuild_package
+            def default_message(tag_name)
+                if tag_name
+                    "autoproj created tag #{tag_name}"
+                else
+                    'autoproj created version commit'
+                end
+            end
+
+            def run(arguments, options = Hash.new)
+                tag_name, *user_selection = *arguments
+                ws.load_config
+                pkg = ws.manifest.main_package_set.create_autobuild_package
                 importer = pkg.importer
                 if !importer || !importer.kind_of?(Autobuild::Git)
-                    raise CLIInvalidArguments, "cannot use autoproj tag if the main configuration is not managed by git"
+                    raise CLIInvalidArguments, "cannot use autoproj commit if the main configuration is not managed by git"
                 end
 
-                versions_file = File.join(
-                    ws.config_dir,
-                    Workspace::OVERRIDES_DIR,
-                    Versions::DEFAULT_VERSIONS_FILE_BASENAME)
+                unless tag_name.nil?
```

Comment:
This is `if !tag_name.nil?`. A.k.a. `if tag_name`.

Two style constraints in packages I maintain:
- don't use `nil?` unless you really mean it. In this case, the "thruthiness" of tag_name is enough
- don't use unless, ever.

Reply:
> don't use nil? unless you really mean it

I try my best to follow your style when I am working on stuff that are mostly maintained by you alone. This can be really frustrating sometimes.. In this particular case I did it like this because you did it like this yourself [here](https://github.com/rock-core/autoproj/blob/1fdb3230869325f246ec110ff7f24647c530dbbc/lib/autoproj/cli/tag.rb#L24)

> don't use unless, ever.

This one was really an attempt to follow a community-adopted common practice (suggested by Rubocop) but I will remove it.

Reply:
> I try my best to follow your style when I am working on stuff that are mostly maintained by you alone. This can be really frustrating sometimes..

I honestly don't know why there is a `nil?` here. I never do this. Maybe refactoring of code from other people ... Sorry for that one.

> This one was really an attempt to follow a community-adopted common practice (suggested by Rubocop) but I will remove it.

I'm tuning the Rubocop rules to better fit the style in my packages. There are quite a bit of Rubocop rules that are actually not "accepted best practices".

I've been using `unless` for a while, but the logic gets hard to follow ... one ends up with double-negations quite often (like here)

Reply:
OK, let's soften this a bit: ultimately, this is the style autoproj is coded with (mostly) because that's the one I prefer. In the spirit of being welcoming (here, mostly for you), I would not mind adopting the default rubocop ruleset (minus some stuff that is even listed as "both ways are widely accepted in the ruby style guide"). A.k.a. I'm open for discussion here.

Setting up and using rubocop would definitely help with consistency, on both sides. I do understand that it's frustrating to have to guess how the style should be...

Reply:
> I've been using unless for a while, but the logic gets hard to follow ... one ends up with double-negations quite often (like here)

I only favor `unless` over `if not` and I think (can't say for sure) that's what's suggested in ruby's style guide.

> using rubocop would definitely help with consistency, on both sides

👍  ... I've been doing this for a while I the only thing I override is the indentation width (two spaces really ruins readability for me)

> I do understand that it's frustrating to have to guess how the style should be

I'm glad you got exactly what I meant. I have no problems with following someone else's style in situations like these, the problem is when as you very well said it basically becomes guess work.

Reply:
> There are quite a bit of Rubocop rules that are actually not "accepted best practices".

We should probably open issues in Rubocop's repo for those cases.

```
RuboCop is a Ruby static code analyzer. Out of the box it will enforce many of the guidelinesoutlined in the community Ruby Style Guide.
```

Reply:
> I've been doing this for a while I the only thing I override is the indentation width (two spaces really ruins readability for me)

I started doing it the other day, but then vscode became horribly sluggish. There's a PR on vscode-ruby to rate-limit the calls to rubocop, but it's not merged/released yet.

Reply:
> > There are quite a bit of Rubocop rules that are actually not "accepted best practices".

> We should probably open issues in Rubocop's repo for those cases.

Regardless of what the rubocop guys say, things like the `if !` vs unless is more split than they acknowledge.

There are quite a few rules where they say themselves than any settings for a rule are widely accepted.